### DISCUSSION TITLE
OnPointerDragged() MRTK 3 implementation

### DISCUSSION THREAD

**Discussion by camroo29** [microsoft/MixedRealityToolkit-Unity#11760] [Q&A] [answered]
Hi, 
 
I want to implement the ` public void OnPointerDragged(MixedRealityPointerEventData eventData)` in MRTK 3. What is the alternative of this event function in MRTK 3? I went through the migration docs but it is unclear. Thanks.

**Comment by AMollis** [comment, marked as answer]
@camroo29, you'll need to read up on Unity's XRI package.   All interactions in MRTK3 go through this input system, which is sufficiently different than MRTK2.

One approach for handling drag in Unity's XRI is.

1. Create a component that extends from XRBaseInteractable.
2. Component overrides [XRBaseInteractable::onSelectEntered](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.0/api/UnityEngine.XR.Interaction.Toolkit.XRBaseInteractable.html#UnityEngine_XR_Interaction_Toolkit_XRBaseInteractable_onSelectEnter) and tracks when item is selected.
3. Component overrides [XRBaseInteractable::ProcessInteractable](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.0/api/UnityEngine.XR.Interaction.Toolkit.XRBaseInteractable.html#UnityEngine_XR_Interaction_Toolkit_XRBaseInteractable_ProcessInteractable_UnityEngine_XR_Interaction_Toolkit_XRInteractionUpdateOrder_UpdatePhase_), and components the drag amount in this function  (see MRTK3's [ObjectManipulator::ProcessInteractable ](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/c86b0e95bdc3bb5b40f60fe9b2571a1b55b0402c/com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs#L730)for an example).
4. Component overrides [XRBaseInteractable::onSelectExited](https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.0/api/UnityEngine.XR.Interaction.Toolkit.XRBaseInteractable.html#UnityEngine_XR_Interaction_Toolkit_XRBaseInteractable_OnSelectExited_UnityEngine_XR_Interaction_Toolkit_SelectExitEventArgs_) and tracks when item is unselected. 

I highly recommend reading up on https://docs.unity3d.com/Packages/com.unity.xr.interaction.toolkit@2.4/manual/index.html

**Reply by camroo29** [reply]
Thanks @AMollis
